tensorflow-ops/src/TensorFlow/Gradient.hs

```diff
@@ -402,6 +403,21 @@ type GradientFunc a = NodeDef
 toT :: Output -> Tensor Value a
 toT = Tensor ValueKind
 
+
+-- | Wrapper around `TensorFlow.GenOps.Core.slice` that builds vectors from scalars for
+-- simple slicing operations.
+flatSlice :: forall v1 t i . (TensorType t, OneOf '[Int32, Int64] i)
```

Comment:
You seem to only need the Int32 version, let's reduce generality and improve call sites?